packages/components/src/number-control/index.tsx

```diff
@@ -35,32 +38,36 @@ export function NumberControl(
 		type: typeProp = 'number',
 		value: valueProp,
 		...props
-	},
-	ref
+	}: WordPressComponentProps< NumberControlProps, 'input', false >,
+	ref: ForwardedRef< any >
 ) {
 	const isStepAny = step === 'any';
+	// @ts-expect-error step should be a number but could be string
```

Comment:
Perfect, thanks! I replaced all the `parseFloat`s, but I'll pass on the `ensureString` for this PR since that hinges on the "what type should `value` be" decision, and it's dangerous enough to affect unit test results 😄